Automattic/wp-calypso
# Domains: Add advanced transfer options card

### Changes proposed in this Pull Request

**Note**: This depends on D70720-code.

This PR creates the advanced transfer options card in the new redesigned transfer page. This card contains a transfer lock toggle for the domain and a button to send the domain's authorization code to its owner - these actions were previously contained in the "Transfer to another registrar" option, which we won't have anymore.

A new action and reduce were created because the existing `DOMAIN_TRANSFER_CODE_REQUEST` one encompassed unlocking the domain and also disabling its privacy and sending the auth code via email, which will remain as separate actions now.

If the domain is currently in a transfer lock - either because it's a new domain (less than 60 days old) or because its contact info was updated and the owner chose not to opt-put of the lock - the toggle will be disabled and an explanation and a link to the relevant support docs will be shown to the user.

This is part of the domain management redesign project described in pcYYhz-m2-p2.

#### Screenshots

Old "Transfer to another registrar" page that will be removed

<img width="770" alt="Screen Shot 2021-11-26 at 15 11 35" src="https://user-images.githubusercontent.com/5324818/143618546-d68057f0-dc3a-4c39-83f4-84bafffbd033.png">

Advanced options card in desktop view

![Markup on 2021-12-08 at 16:04:22](https://user-images.githubusercontent.com/5324818/145268517-ec6f4aa7-c1d8-436c-9bdb-597c6025a0ae.png)

And in mobile view

![Markup on 2021-12-08 at 16:05:11](https://user-images.githubusercontent.com/5324818/145268525-9f518056-2a96-402b-8101-a400ec658c90.png)

When the domain is under a transfer lock, the following message will be displayed:

<img width="741" alt="Screen Shot 2021-12-08 at 16 03 34" src="https://user-images.githubusercontent.com/5324818/145269008-d9e860ad-7a35-47fc-8c30-95d7ab730f8a.png">

### Testing instructions

- Apply D70720-code to your sandbox if it hasn't been merged yet and sandbox `public-api`
- Build this branch locally or open the live Calypso link
- Go to "Upgrades > Domains"
- Select one of your registered domains or domain connections
- If you're in the live Calypso link, please append ?flags=domains/transfers-redesign to your URL to enable the appropriate feature flag
- Select the "Transfer your domain" options
- Please check that:
  - Advanced options card is present
  - Lock/unlock toggle works as expected
  - "Get authorization code" button works
  - Mobile view looks correct as shown in the screenshot above
  - If you selected a domain connection, ensure the advanced options do not appear, as they are only pertinent to registered domains
  - Also please check a domain which is under transfer lock
